api/src/main/resources/liquibase.xml

```diff
@@ -712,4 +712,23 @@
             ALTER TABLE patient_appointment MODIFY end_date_time datetime default null;
         </sql>
     </changeSet>
+    <changeSet id="add-patient-appointment-fulfilling-encounter-map-table-20240501" author="mgoodrich">
+        <preConditions onFail="MARK_RAN">
+            <not>
+                <tableExists tableName="patient_appointment_fulfilling_encounter_map" />
+            </not>
+        </preConditions>
+        <comment>Add the map table to link appointments to fulfilling encounter</comment>
+        <createTable tableName="patient_appointment_fulfilling_encounter_map">
```

Comment:
should we have "dateCreated", "createdBy" columns as well?

Reply:
It's a good question @angshu .  My original take on this was that this is not really an independent domain object, but more to support a Collection property on Encounter and Appointment so we don't really need them.  i.e. this is essentially here to support both a Set<Encounter> property on appointment, and a (virtual) Set<Appointment> property on Encounter.  But if the typical pattern is to include Auditable properties on objects like this, or if there are known use cases that require it, then we can do that.

Reply:
Yeah, @mseaton  and I went back and forth on this a bit and decided that having auditable fields on this table probably wasn't needed for the reasons Mike suggests.

I'd be inclined to keep it is unless you feel strongly it's needed @angshu ?  We'd also need to build in proper cascading voidng (ie when you void the appointment you should void the appointment_encounter_map), and we'd want to build full void support into the Appointments Service, as that doesn't appear to be 100% implemented now.

Reply:
no, I am ok. thank you